src/OpenTracing/SpanOptions.php

```diff
@@ -43,7 +43,17 @@ public static function create(array $options)
 
                 case 'tags':
                     foreach ($value as $tag => $tagValue) {
-                        $spanOptions->tags[] = Tag::create($tag, $tagValue);
+                        if ($tag !== (string) $tag) {
+                            throw InvalidSpanOption::invalidTag($tag);
+                        }
+
+                        if (!is_scalar($tagValue)
+                            && !(is_object($tagValue) && method_exists($tagValue, '__toString'))
```

Comment:
To clarify, what I mean is that only user input should be validated. This is programmer input, and it is documented what types are accepted, so I would trust the programmer to only pass valid types (or ensure valid types himself if he passes user input). The VO code also validated the input when using the key/value methods or validated that the array consists of VO objects, so I don't think this is a point in favour of them.